backend/Dockerfile

```diff
@@ -15,9 +15,11 @@ COPY backend/requirements.txt .
 RUN python3 -m pip install -r requirements.txt --no-cache-dir
 
 # Downloading Argo CLI so that the samples are validated
-#ADD  https://github.com/argoproj/argo/releases/download/v2.7.5/argo-linux-amd64 /usr/local/bin/argo
-ADD  https://github.com/argoproj/argo/releases/download/v2.4.3/argo-linux-amd64 /usr/local/bin/argo
-RUN chmod +x /usr/local/bin/argo
+ENV ARGO_VERSION v2.12.9
```

Comment:
Does the CLI work offline again?

Reply:
Not sure that it is an issue depending on what we need to do offline,  https://github.com/argoproj/argo-workflows/issues/2793#issuecomment-772633106. : 
```
You can lint offline, as long as you do not use template references.
```

do we depend on [template references](https://github.com/argoproj/argo-workflows/blob/master/docs/workflow-templates.md) and do we use the argo cli for other use cases than validating(lint) workflows?